src/frontend/src/optimizer/plan_node/generic/source.rs

```diff
@@ -49,20 +49,22 @@ pub struct Source {
 
 impl GenericPlanNode for Source {
     fn schema(&self) -> Schema {
-        let fields = self.non_generated_columns().map(Into::into).collect();
+        let fields = self
+            .column_catalog
+            .iter()
+            .map(|c| (&c.column_desc).into())
+            .collect();
         // let fields = self.column_descs.iter().map(Into::into).collect();
         Schema { fields }
     }
 
     fn logical_pk(&self) -> Option<Vec<usize>> {
         let mut id_to_idx = HashMap::new();
-        // self.column_descs.iter().filter(|c| !c.is_generated_column()).enumerate().for_each(|(idx,
+        // self.column_descs.iter().filter(|c| !c.is_generated()).enumerate().for_each(|(idx,
         // c)| {
-        self.non_generated_columns()
-            .enumerate()
-            .for_each(|(idx, c)| {
-                id_to_idx.insert(c.column_id, idx);
-            });
+        self.column_catalog.iter().enumerate().for_each(|(idx, c)| {
+            id_to_idx.insert(c.column_id(), idx);
+        });
```

Comment:
Should we return `None` here considering a source shouldn't have pk?

Reply:
No, `SourceExecutor` does not need to be aware of pk, while it is useful in frontend plan generation.